api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricHandler.java

```diff
@@ -101,20 +107,34 @@ public void findMetrics(
     @ApiOperation(value = "Add data for multiple metrics in a single call.")
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Adding data succeeded."),
+            @ApiResponse(code = 400, message = "Missing or invalid payload.", response = ApiError.class),
             @ApiResponse(code = 500, message = "Unexpected error happened while storing the data",
                 response = ApiError.class) })
     public void addMetricsData(
             @Suspended final AsyncResponse asyncResponse,
             @ApiParam(value = "List of metrics", required = true) MixedMetricsRequest metricsRequest) {
 
-        Observable<Void> gaugeInserts = metricsService.addGaugeData(requestToGauges(tenantId,
-                metricsRequest.getGaugeMetrics()).subscribeOn(Schedulers.computation()));
-        Observable<Void> counterInserts = metricsService.addCounterData(requestToCounters(tenantId,
-                metricsRequest.getCounters()).subscribeOn(Schedulers.computation()));
-        Observable<Void> availabilityInserts = metricsService.addAvailabilityData(requestToAvailabilities(tenantId,
-                metricsRequest.getAvailabilityMetrics()).subscribeOn(Schedulers.computation()));
+        Collection<Observable<Void>> observables = new ArrayList<>();
```

Comment:
This isn't very reactive design and actually a bit more difficult to follow what's happening. Having empty observables is not an issue, they are just not processed (nothing to react to).